### PR TITLE
Add `ActorBase` lifecycle flow unit tests

### PR DESCRIPTION
**Pull request by Arkatufus**
## Changes

* Add new property `ActiveTimers` to `ITimerScheduler` to retrieve all currently active timer keys
* Adds `ActorBase` lifecycle flow unit tests in relation to `IWithTimers` lifecycle

## Checklist

For significant changes, please ensure that the following have been completed (delete if not relevant):

* [x] This change follows the [Akka.NET API Compatibility Guidelines](https://getakka.net/community/contributing/api-changes-compatibility.html).
* [ ] I have [reviewed my own pull request](https://getakka.net/community/contributing/index.html#review-your-own-pull-requests).
* [ ] Changes in public API reviewed, if any.

### CODE REVIEW

File: src/core/Akka/Actor/Scheduler/TimerScheduler.cs [hunk 1]
```diff
@@ -150,6 +151,11 @@ public bool IsTimerActive(object key)
             return _timers.ContainsKey(key);
         }
 
+        /// <summary>
+        /// Retrieves all current active timer keys
+        /// </summary>
+        public ImmutableArray<object> ActiveTimers => _timers.Keys.ToImmutableArray();
```
Comment (Aaronontheweb) [inline src/core/Akka/Actor/Scheduler/TimerScheduler.cs]: I would just return an `ICollection<object>` or whatever here so we don't have to allocate an immutable array each time.

**Review by Aaronontheweb — CHANGES_REQUESTED** [review]
need to make the read operation zero-allocation.

File: src/core/Akka/Actor/Scheduler/TimerScheduler.cs (hunk 1 above)
Comment (Aaronontheweb) [reply src/core/Akka/Actor/Scheduler/TimerScheduler.cs]: I understand that this is probably done for the purposes of making sure we don't get rugged by the collection changing out from under us, but it's better to take that risk and leave the API un-opionated

Comment (Arkatufus) [reply src/core/Akka/Actor/Scheduler/TimerScheduler.cs]: Done, though accessing `_timers.Keys` actually allocates internally inside the `Dictionary`, but that's unavoidable.

**Review by Aaronontheweb — APPROVED** [review]
LGTM

**Pull request merged**